### PR TITLE
remove panics from connect graph

### PR DESCRIPTION
**Pull request by nicholascioli**
This commit removes all potential panics from the connector graph creation logic.

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-federation/src/sources/connect/query_graph.rs [hunk 1]
```diff
@@ -150,7 +152,9 @@ fn process_selection(
             _ => {
                 // If we don't have either of the above, then we must have a subselection
                 let Some(sub) = path.next_subselection() else {
-                    todo!("handle error");
+                    return Err(FederationError::internal(
+                        "expected subselection for non-unit type",
```
Comment (lennyburdette) [inline apollo-federation/src/sources/connect/query_graph.rs:156]: meganit: in graphql parlance these are leaf (versus composite) types

File: apollo-federation/src/sources/connect/query_graph.rs [hunk 2]
```diff
@@ -190,29 +196,14 @@ fn process_subselection(
     node_cache: &mut IndexMap<Name, NodeIndex<u32>>,
     properties_path: Option<Vec<Property>>,
 ) -> Result<NodeIndex<u32>, FederationError> {
-    // Reference for working with the entry API
```
Comment (lennyburdette) [inline apollo-federation/src/sources/connect/query_graph.rs:193]: do you have a TODO for using the cache for leaf types?

File: apollo-federation/src/sources/connect/query_graph.rs [hunk 3]
```diff
@@ -239,7 +230,10 @@ fn process_subselection(
         // Make sure that we have a field on the object type that matches the alias (or the name itself)
         let alias = selection.name();
         let Some(selection_field) = object_type.fields.get(alias) else {
-            todo!("handle error");
+            return Err(FederationError::internal(format!(
+                "expected alias `{alias}` to exist in GraphQL object `{}`",
```
Comment (lennyburdette) [inline apollo-federation/src/sources/connect/query_graph.rs:234]: aliases are a GraphQL operation thing, not a GraphQL type system thing. you could say "expected field `{alias}` to exist on GraphQL type `{}`"

**Review by lennyburdette — APPROVED** [review]

File: apollo-federation/src/sources/connect/query_graph.rs (hunk 2 above)
Comment (nicholascioli) [reply apollo-federation/src/sources/connect/query_graph.rs:193]: This comment was mostly here as reference. The existing code already uses the cache

**Pull request merged**